battery_management_system_test.py

```diff
@@ -0,0 +1,40 @@
+import check_battery_limits as checker
+
+if __name__ == '__main__':
```

Comment:
Comprehensive tests :)
But couldn't see a test where `collect_out_of_range_battery_parameters` returns an empty array. How do you ensure that all variants are tested?